### DISCUSSION TITLE
Template files

### DISCUSSION THREAD

**Discussion by ClemensElflein** [xtech/self-o-mat#12] [General]
@maehw,
I have seen that you started documenting the files used by the software in the wiki.

For the template:
>    /.template.png: ? accessed by ImageProcessor::loadTemplateImage()
>   /.template_props.json: ? accessed by ImageProcessor::loadTemplateImage() and ImageProcessor::updateTemplate()
>   /.template_screen.png: ? accessed by ImageProcessor::updateTemplate() and BoothGui::reloadTemplate()
>   /.template_screen_props.json: ? acessed by ImageProcessor::updateTemplate() and BoothGui::reloadTemplate()

All 4 files will be created by the software as soon as a template is uploaded using the website. The two .png files are the images (the first is the full resolution template, the _screen is the template scaled down for the UI. This allows faster rendering since we don't need to scale it later). The two _prop.json files contain the rectangle where the image needs to be inserted in px coordinates. Again there is a file for the final print file as well as for the low resolution screen preview.

Hope that helps

**Comment by maehw** [comment]
Are the original photos from the camera preserved as well or are only the overlayed/printout photos stores?
IIRC there was a way to store both - but I am not sure.

**Reply by ClemensElflein** [reply]
Both versions should be stored, since we didn't want to throw away the originals.
